src/Configuration.cpp

```diff
@@ -112,6 +114,10 @@ void Configuration::assign_by_commandline(){
   transform_loop_unroll = cl_transform_loop_unroll;
   print_progress = cl_print_progress || cl_print_progress_estimate;
   print_progress_estimate = cl_print_progress_estimate;
+  program_arguments.push_back("prog");
+  for(std::string a : cl_program_arguments){
```

Comment:
You could use `vector.insert` here; but I admit that's a taste thing.

Reply:
i will change to vector.insert.